src/ui/qgsprojectpropertiesbase.ui

```diff
@@ -1802,6 +1802,13 @@
                 </property>
                </widget>
               </item>
+              <item row="1" column="0">
+               <widget class="QCheckBox" name="mAsynchProviderCreationCheckBox">
+                <property name="text">
+                 <string>Allow asynchronous provider creation on project loading</string>
```

Comment:
How about renaming to this:

Load map layers in parallel when loading project. This may significantly speed up loading of projects (but only some providers take advantage of this).